corehq/apps/locations/views.py

```diff
@@ -750,17 +752,42 @@ def products_at_location(self):
 
     @property
     def page_name(self):
-        return mark_safe(_("Edit {name} <small>{type}</small>").format(
+        if self.request.is_view_only:
+            name = _("View {name} <small>{type}</small>")
+        else:
+            name = _("Edit {name} <small>{type}</small>")
+        return mark_safe(name.format(
             name=self.location.name, type=self.location.location_type_name
         ))
 
+    @property
+    def can_edit_commcare_users(self):
+        return self.request.couch_user.can_edit_commcare_users()
+
+    @property
+    def can_view_commcare_users(self):
+        return self.request.couch_user.can_view_commcare_users()
+
     @property
     def page_context(self):
         context = super(EditLocationView, self).page_context
+
+        if self.request.is_view_only:
+            make_form_readonly(self.location_form.location_form)
+            make_form_readonly(self.location_form.custom_location_data.form)
+            make_form_readonly(self.products_form)
+            make_form_readonly(self.related_location_form)
+            if not self.can_edit_commcare_users:
```

Comment:
Why is this condition here? If the whole request is read only, they shouldn't be able to edit anything, right?

Reply:
if they can edit commcare users but can only view locations, they can still change what location a user is assigned to, which is what this does